scripts/network_bandwidth.sh

```diff
@@ -1,51 +1,68 @@
-    final_upload=$(cat /sys/class/net/$network_name/statistics/tx_bytes)
-
-    total_download_bps=$(expr $final_download - $initial_download)
-    total_upload_bps=$(expr $final_upload - $initial_upload)
-
-    if [ $total_download_bps -gt 1073741824 ]; then
-      output_download=$(echo "$total_download_bps 1024" | awk '{printf "%.2f \n", $1/($2 * $2 * $2)}')
-      output_download_unit="gB/s"
-    elif [ $total_download_bps -gt 1048576 ]; then
-      output_download=$(echo "$total_download_bps 1024" | awk '{printf "%.2f \n", $1/($2 * $2)}')
-      output_download_unit="mB/s"
-    else
-      output_download=$(echo "$total_download_bps 1024" | awk '{printf "%.2f \n", $1/$2}')
-      output_download_unit="kB/s"
-    fi
-
-    if [ $total_upload_bps -gt 1073741824 ]; then
-      output_upload=$(echo "$total_download_bps 1024" | awk '{printf "%.2f \n", $1/($2 * $2 * $2)}')
-      output_upload_unit="gB/s"
-    elif [ $total_upload_bps -gt 1048576 ]; then
-      output_upload=$(echo "$total_upload_bps 1024" | awk '{printf "%.2f \n", $1/($2 * $2)}')
-      output_upload_unit="mB/s"
-    else
-      output_upload=$(echo "$total_upload_bps 1024" | awk '{printf "%.2f \n", $1/$2}')
-      output_upload_unit="kB/s"
-    fi
-
-    echo "↓ $output_download $output_download_unit • ↑ $output_upload $output_upload_unit"
-  done
+current_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source $current_dir/utils.sh
+
+get_bytes()
+{
+  case $(uname -s) in
+    Linux)
+      download=$(cat /sys/class/net/$1/statistics/rx_bytes)
+      upload=$(cat /sys/class/net/$1/statistics/tx_bytes)
+      echo "$download $upload"
+      ;;
+
+    Darwin)
+      # TODO - Darwin/Mac compatability
+      ;;
+
+    CYGWIN*|MINGW32*|MSYS*|MINGW*)
+      # TODO - windows compatability
+      ;;
+  esac
+}
+
+format_bytes()
+{
+  if [ $1 -gt 1073741824 ]; then
+    format_num=$(echo "$1 1024" | awk '{printf "%.2f \n", $1/($2 * $2 * $2)}')
+    echo "$format_num gB/s"
+  elif [ $1 -gt 1048576 ]; then
+    format_num=$(echo "$1 1024" | awk '{printf "%.2f \n", $1/($2 * $2)}')
+    echo "$format_num mB/s"
+  else
+    format_num=$(echo "$1 1024" | awk '{printf "%.2f \n", $1/$2}')
+    echo "$format_num kB/s"
+  fi
 }
+
+get_bandwidth() {
+  read initial_download initial_upload < <(get_bytes $2)
+
+  sleep $1
+
+  read final_download final_upload < <(get_bytes $2)
+
+  total_download_bps=$(echo "$final_download $initial_download $1" | awk '{printf "%.0f \n", ($1 - $2) / $3}')
+  total_upload_bps=$(echo "$final_upload $initial_upload $1" | awk '{printf "%.0f \n", ($1 - $2) / $3}')
+
+  output_download=$(format_bytes $total_download_bps)
+  output_upload=$(format_bytes $total_upload_bps)
+
+  echo "↓ $output_download • ↑ $output_upload"
+}
+
+main()
+{
+  # storing the refresh rate in the variable RATE, default is 5
+  RATE=$(get_tmux_option "@dracula-refresh-rate" 5)
+  default_network_name=$(ip route | grep default | sed -e "s/^.*dev.//" -e "s/.proto.*//")
```

Comment:
That will break on MacOS, `ip` is not a command. This should probably belong in its own function.

Reply:
I fix it and also add bandwidth read for MacOS~